contracts/LinearCodeAddressGenerator.cdc

```diff
@@ -0,0 +1,105 @@
+
+/// LinearCodeAddressGenerator allows generating and validating Flow account addresses.
+access(all)
+contract LinearCodeAddressGenerator {
+
+    access(all)
+    let codeWords: {String: UInt64}
+
+    /// Rows of the generator matrix G of the [64,45]-code used for Flow addresses.
+    /// G is a (k x n) matrix with coefficients in GF(2), each row is converted into
+    /// a big endian integer representation of the GF(2) raw vector.
+    /// G is used to generate the account addresses
+    access(all)
+    let generatorMatrixRows: [UInt64; 45]
+
+    /// Columns of the parity-check matrix H of the [64,45]-code used for Flow addresses.
+    /// H is a (n x p) matrix with coefficients in GF(2), each column is converted into
+    /// a big endian integer representation of the GF(2) column vector.
+    /// H is used to verify a code word is a valid account address.
+    access(all)
+    let parityCheckMatrixColumns: [UInt64; 64]
+
+    init() {
+        self.codeWords = {
```

Comment:
these are invalid code words rather than code words, maybe it's better to add the "invalid" prefix for clarity.

Reply:
I found that terminology really confusing from a user-perspective. Though as far as I understand the codewords are technically invalid and mixed into the value, they are the values that should be used by the user. If a user sees "invalid" they think "should not use". Maybe we can hide this implementation detail?

Reply:
You're right, there is only one space of valid code words (which is the address space we use for mainnet). 
To get a different disjoint space of addresses, we get the valid code word space and mix it with an invalid value (value that is not a code word). For each invalid value, we get a new space of addresses (none of the new space addresses is a valid code word).
The fact that the value is invalid is important to get disjoint spaces of addresses, though I agree it is an implementation detail. Users should only use addresses and shouldn't care about code words.

On the Go side, code words are hidden from the package users. On Cadence, I'm not sure how the contract is going to be used. I don't think any user should access these values directly so they can be hidden (made private, non exported maybe?) to avoid confusion, same as the rest of the linear codes stuff. 
However, since the contract is using correct names for the linear code things (matrix, parity, linearity..), it makes sense IMO to keep the mention that these are invalid code words.

Reply:
I've hid all the implementation details (matrix, partity, etc.) and introduced the notion of a chain in [26ba620](https://github.com/onflow/flow-core-contracts/pull/467/commits/26ba6207e7feb0d72d8c599759ffa51eef50a5ce). So the only public API is now:

- `Chain` type
- `fun address(at index: UInt64, chain: Chain): Address`
- `fun isValidAddress(_ address: Address, chain: Chain): Bool`